src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Options/PackageGenCommonOptions.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using CommandLine;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Exceptions;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Cli.Options
+{
+    internal abstract class PackageGenCommonOptions
+    {
+        protected PackageGenCommonOptions(
+            string sourceDirectory,
+            bool overwrite,
+            string? manifestFilePath)
+        {
+            this.SourceDirectory = sourceDirectory;
+            this.Overwrite = overwrite;
+            this.ManifestFilePath = manifestFilePath;
+        }
+
+        [Option(
+           's',
+           "source",
+           Required = true,
+           HelpText = "The directory containing the plugin binaries.")]
+        public string SourceDirectory { get; }
+
+        [Option(
+            'w',
+            "overwrite",
+            Required = false,
+            HelpText = "Indicates that the destination file should be overwritten if it already exists.")]
+        public bool Overwrite { get; } = false;
+
+        [Option(
+            'm',
+            "manifest",
+            Required = false,
+            HelpText = "Path to the plugin manifest file. If not specified, the program will attempt to find the manifest in the source directory.")]
+        public string? ManifestFilePath { get; }
+
+        internal string SourceDirectoryFullPath { get; private set; }
+
+        internal string? ManifestFileFullPath { get; private set; }
```

Comment:
IMO, it's better to separate out the "parse result" with "actual representation the app wants to work with" into two separate classes. The idea being you have a single class whose only job is to contain the parsed result, Then you have a way to transform that class into whatever representation you actually want that have these above properties